component/mimir/rules/diff.go

```diff
@@ -0,0 +1,111 @@
+package rules
+
+import (
+	"bytes"
+
+	mimirClient "github.com/grafana/agent/pkg/mimir/client"
+
+	"gopkg.in/yaml.v3"
```

Comment:
We typically try to use yaml.v2 (though at this point I don't remember why). Is v3 necessary here?

Reply:
v3 is necessary because its used to unmarshal into structs that directly reference v3 types. For example the `RuleNode` type in prometheus: https://github.com/grafana/prometheus/blob/master/pkg/rulefmt/rulefmt.go#L124. Prometheus uses the yaml Node types directly in order to provide line numbers for syntax and validation errors.

I've tested it with v2, which fails to unmarshal.

Reply:
I'll add a small comment to that line to clarify